sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -287,23 +291,31 @@ def to_arrow(self) -> pyarrow.Table:
 
 
 def block_until_done(client, bq_job):
-    def _is_done(job_id):
-        return client.get_job(job_id).state in ["PENDING", "RUNNING"]
+    """
+    Waits a maximum of 30 minutes for bq_job to finish running.
+    Args:
+        client: A bigquery.client.Client to monitor the bq_job.
+        bq_job: The bigquery.job.QueryJob that blocks until done runnning.
+    Raises:
+        BigQueryJobStillRunning exception if the function has blocked longer than 30 minutes.
+        BigQueryJobCancelled exception on a KeyboardInterrupt.
+    """
 
     @retry(wait=wait_fixed(10), stop=stop_after_delay(1800), reraise=True)
     def _wait_until_done(job_id):
-        return _is_done(job_id)
+        if client.get_job(job_id).state in ["PENDING", "RUNNING"]:
+            raise BigQueryJobStillRunning(job_id=job_id)
 
     job_id = bq_job.job_id
-    _wait_until_done(job_id=job_id)
+    try:
+        _wait_until_done(job_id=job_id)
+    except KeyboardInterrupt:
+        client.cancel_job(job_id)
+        raise BigQueryJobCancelled(job_id=job_id)
```

Comment:
Should you also have a 

```python
except tenacity.RetryError as e:
   raise e
```
?

Reply:
This is what happened when my 10 seconds timeout hit
<img width="1074" alt="image" src="https://user-images.githubusercontent.com/18557047/125528786-e6c0bfcc-2b9c-4dce-b19e-c2427f51db1a.png">

Reply:
Isn't it the same behavior with or without the second except clause?
```python
except tenacity.RetryError as e:
   raise e
```

I believe the try and except behavior (see [Python docs](https://docs.python.org/3/tutorial/errors.html#handling-exceptions)) will cause any other exception not named in the except clause to be raised as it would've:
> If an exception occurs which does not match the exception named in the except clause, it is passed on to outer try statements; if no handler is found, it is an unhandled exception and execution stops with a message as shown above.

Reply:
Lol I am so stupid 🤦 
Sorry for giving you hard times with this PR, yesterday was wild.

Will have a final pass but it will still need to be approved by a Feast dev

Reply:
No worries at all Matt! Thanks for all the time spent poking at this - just want to make sure things make sense.